### PR TITLE
ipatool 2.1.4 (new formula)

### PR DESCRIPTION
**Pull request by zyoshoka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/i/ipatool.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  def install
+    system "go", "build", "-o", bin/"ipatool"
+
+    generate_completions_from_executable(bin/"ipatool", "completion")
+  end
+
+  test do
+    output = shell_output("#{bin}/ipatool auth info 2>&1", 1)
+    assert_match "failed to get account", output
+  end
```
Comment (chenrui333) [inline Formula/i/ipatool.rb:19]: can you also assert the version as well?

File: Formula/i/ipatool.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  desc "CLI tool for searching and downloading app packages from the iOS App Store"
+  homepage "https://github.com/majd/ipatool"
+  url "https://github.com/majd/ipatool/archive/refs/tags/v2.1.4.tar.gz"
+  sha256 "e0e01c88efb94f35a71f664267c6c9ab0e22932804e0af864a0a5cd8d348dbca"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X github.com/majd/ipatool/v2/cmd.version=#{version}"), "-o", bin/"ipatool"
```
Comment (SMillerDev) [inline Formula/i/ipatool.rb:11]: This defaults to the formula name, so this should work without.
```suggestion
    system "go", "build", *std_go_args(ldflags: "-X github.com/majd/ipatool/v2/cmd.version=#{version}")
```

File: Formula/i/ipatool.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    generate_completions_from_executable(bin/"ipatool", "completion")
```
Comment (SMillerDev) [inline Formula/i/ipatool.rb:13]: I believe this is the default
```suggestion
    generate_completions_from_executable
```

Comment (zyoshoka) [reply Formula/i/ipatool.rb]: Seems that it is not correct.

https://github.com/Homebrew/homebrew-core/actions/runs/11995086748/job/33437987406

File: Formula/i/ipatool.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  desc "CLI tool for searching and downloading app packages from the iOS App Store"
+  homepage "https://github.com/majd/ipatool"
+  url "https://github.com/majd/ipatool/archive/refs/tags/v2.1.4.tar.gz"
+  sha256 "e0e01c88efb94f35a71f664267c6c9ab0e22932804e0af864a0a5cd8d348dbca"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X github.com/majd/ipatool/v2/cmd.version=#{version}")
```
Comment (chenrui333) [inline Formula/i/ipatool.rb:11]: that would help reduce the binary size

```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/majd/ipatool/v2/cmd.version=#{version}")
```

**Review by chenrui333 — APPROVED** [review]
besides ldflag change, others lgtm

**Review by daeho-ro — APPROVED** [review]

**Pull request merged**